prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,951 @@
+            if count > 0:
+                sum_kv = sum(kv_dict[nei] for nei in neighbors[depth] if nei in kv_dict)
+                kv_dict[sub] = sum_kv / count
+                break
+
+
+def set_sub(clean_data):
+    """Generate the subs
+
+    :param str E_csv: path of the HIFLD substation csv file
+    :return: (*dict*) --  sub_by_coord_dict, a dict mapping from (x, y) to substation detail.
+    :return: (*dict*) --  sub_name_dict, a dict mapping from substation name to its coordinate (x, y).
+    """
+
+    sub_by_coord_dict = {}
+    sub_name_dict = {"sub": []}
+    for index, row in clean_data.iterrows():
+        location = (row["LATITUDE"], row["LONGITUDE"])
+        if location in sub_by_coord_dict:
+            raise Exception(
+                f"WARNING: substations coordinates conflict check: {location}"
+            )
+        sub_by_coord_dict[location] = (
+            row["ID"],
+            row["NAME"],
+            row["STATE"],
+            row["COUNTY"],
+        )
+        if row["NAME"] not in sub_name_dict:
+            sub_name_dict[row["NAME"]] = []
+        sub_name_dict[row["NAME"]].append((row["LATITUDE"], row["LONGITUDE"]))
+    return sub_by_coord_dict, sub_name_dict
+
+
+def write_sub(clean_data, zone_dic, zone_dic1, region):
+    """Write the data to sub.csv as output
+
+    :param pandas.DataFrame clean_data: substation dataframe as returned by :func:`Clean`
+    :param dict zone_dic: zone dict as returned by :func:`get_Zone`
+    """
+    tx_west = ["EL PASO", "HUDSPETH"]
+    tx_east = [
+        "BOWIE",
+        "MORRIS",
+        "CASS",
+        "CAMP",
+        "UPSHUR",
+        "GREGG",
+        "MARION",
+        "HARRISON",
+        "PANOLA",
+        "SHELBY",
+        "SAN AUGUSTINE",
+        "SABINE",
+        "JASPER",
+        "NEWTON",
+        "ORANGE",
+        "JEFFERSON",
+        "LIBERTY",
+        "HARDIN",
+        "TYLER",
+        "POLK",
+        "TRINITY",
+        "WALKER",
+        "SAN JACINTO",
+        "DALLAM",
+        "SHERMAN",
+        "HANSFORD",
+        "OCHLTREE" "LIPSCOMB",
+        "HARTLEY",
+        "MOORE",
+        "HUTCHINSON",
+        "HEMPHILL",
+        "RANDALL",
+        "DONLEY",
+        "PARMER",
+        "BAILEY",
+        "LAMB",
+        "HALE",
+        "COCHRAN",
+        "HOCKLEY",
+        "LUBBOCK",
+        "YOAKUM",
+        "TERRY",
+        "LYNN",
+        "GAINES",
+    ]
+
+    sd_west = ["LAWRENCE", "BUTTE", "FALL RIVER"]
+    # nm_east = ['CURRY', 'LEA', 'QUAY', 'ROOSEVELT', 'UNION']
+    mt_east = [
+        "CARTER",
+        "CUSTER",
+        "ROSEBUD",
+        "PRAIRIE",
+        "POWDER RIVER",
+        "DANIELS",
+        "MCCONE",
+        "DAWSON",
+        "RICHLAND",
+        "FALLON",
+        "GARFIELD",
+        "ROOSEVELT",
+        "PHILLIPS",
+        "SHERIDAN",
+        "VALLEY",
+        "WIBAUX",
+    ]
+    sub = open("output/sub.csv", "w", newline="")
```

Comment:
If we have the paths hard-coded like this, it means that the code is inflexible, and needs to be run from a specific location. Ideally, the user would be able to provide a root path and the files would be populated into this path.